storage/storage.go

```diff
@@ -61,7 +63,43 @@ func (storage Storage) Put(ctx context.Context, url string, path string, class t
 	if err != nil {
 		return nil, err
 	}
-	return backend.Put(ctx, url, path, class)
+
+	var out []*tes.OutputFileLog
+
+	switch class {
+	case File:
+		err = backend.PutFile(ctx, url, path)
+		if err != nil {
+			return nil, err
+		}
+		out = append(out, &tes.OutputFileLog{
+			Url:       url,
+			Path:      path,
+			SizeBytes: fileSize(path),
+		})
+	case Directory:
+		var files []hostfile
+		files, err = walkFiles(path)
+
+		for _, f := range files {
+			u := strings.TrimSuffix(url, "/") + "/" + f.rel
```

Comment:
We lost something here. The local backend was using `path.Join`.

Reply:
This ends up being equivalent.

Reply:
It's not equivalent for windows.

Reply:
I don't think we have windows support anyways

Reply:
We don't need to be making it more difficult by handling file paths the wrong way.

Reply:
technically its a file URL at this point. According to https://en.wikipedia.org/wiki/File_URI_scheme this would still work for windows

Reply:
I can live with that